app.py

```diff
@@ -84,9 +84,7 @@ def filter(self, record):
     for l in other_loggers:
         l.addHandler(cw_handler)
 
-
-if not config.DEVMODE:
-    VALID_TOPICS = config.get_valid_topics()
+VALID_TOPICS = config.get_valid_topics()
```

Comment:
I would take this out of the global namespace either pass it directly into the handler (not sure if thats really possible...i'm not that familiar with tornado) or have the handler call the method directly.  

Actually, i would probably pass in the data or pass in the config module (or class).  Passing things in like that makes for easier unit testing.

Reply:
I dumped this into `config`. I tried the method of calling it but that would have been doing a lot of work every time a new upload was sent. instead we're just creating the variable in config and i'm passing it in from there.

Reply:
That makes sense.

I think this looks like a step in the right direction.

One thing you might consider is to gather up the config data (or other external dependencies) and pass it along to the RequestHandler subclasses.  For example:

valid_topics = config.get_valid_topics()
build_id = config.BUILD_ID
build_date = config.get_commit_date(build_id)

endpoints = [
    (config.API_PREFIX, RootHandler),
    (config.API_PREFIX + "/v1/upload", UploadHandler, dict(valid_topics=valid_topics)),
    (config.API_PREFIX + "/v1/version", VersionHandler, dict(build_id=build_id, build_date=build_date)),
    (config.API_PREFIX + "/v1/openapi.json", SpecHandler),
    (r"/r/insights/platform/ingress", RootHandler),
    (r"/r/insights/platform/ingress/v1/version", VersionHandler, dict(build_id=build_id, build_date=build_date))),
    (r"/r/insights/platform/ingress/v1/upload", UploadHandler, dict(valid_topics=config.get_valid_topics())),
    (r"/r/insights/platform/ingress/v1/openapi.json", SpecHandler),
    (r"/metrics", MetricsHandler)
]

Notice the "dict(...)" additions to the version endpoint and upload endpoint. 

You would need to add an initialize method (see [1]) to the UploadHandler (accept the valid_topics) and VersionHandler (accept the build_id and build_date).  This way you can further decouple the handlers from other external dependencies.  This will make unit testing these bits of the code easier as you can pass in the dependencies.

[1] https://www.tornadoweb.org/en/branch5.1/web.html#entry-points